File: android/app/src/main/kotlin/com/farosos/app/FirebaseAuthSession.kt
package com.farosos.app

import com.google.firebase.Firebase
import com.google.firebase.auth.auth

/**
 * Sesión anónima compartida por los distintos uploaders de Firebase
 * (`FirebaseParticipantUploader` #30, `FirebaseMeshStateUploader` #32) — una
 * sola por instalación (ADR-0003), sin volver a firmarse si ya hay una
 * activa.
 */
object FirebaseAuthSession {
    fun ensureSignedIn(onResult: (Result<Unit>) -> Unit) {
        val auth = Firebase.auth
        if (auth.currentUser != null) {
            onResult(Result.success(Unit))
            return
        }
        auth.signInAnonymously()
            .addOnSuccessListener { onResult(Result.success(Unit)) }
            .addOnFailureListener { error -> onResult(Result.failure(error)) }
    }
}

File: ios/EmergencyApp/Sources/FirebaseAuthSession.swift
import FirebaseAuth
import Foundation

/// Sesión anónima compartida por los distintos uploaders de Firebase
/// (`FirebaseParticipantUploader` #29, `FirebaseMeshStateUploader` #31) — una
/// sola por instalación (ADR-0003), sin volver a firmarse si ya hay una
/// activa.
enum FirebaseAuthSession {
    static func ensureSignedIn(completion: @escaping (Result<Void, Error>) -> Void) {
        if Auth.auth().currentUser != nil {
            completion(.success(()))
            return
        }
        Auth.auth().signInAnonymously { _, error in
            if let error {
                completion(.failure(error))
            } else {
                completion(.success(()))
            }
        }
    }
}
